pkg/v22/resource/clusterconfigmap/desired.go

```diff
@@ -27,10 +33,26 @@ func (r *StateGetter) GetDesiredState(ctx context.Context, obj interface{}) ([]*
 		return nil, microerror.Mask(err)
 	}
 
-	values := map[string]string{
+	// We set the number of replicas to the number of worker nodes. This is set
+	// by the workercount resource using the current number of nodes from the
+	// tenant cluster.
+	ingressControllerReplicas := cc.Status.Worker.Nodes
+	if ingressControllerReplicas == 0 {
+		// If the current number of workers is not set we fallback to using the
+		// desired worker count.
```

Comment:
However scaling to zero would be bad so we fallback to using the desired worker count.